src/i3ipc.ml

```diff
@@ -249,6 +271,23 @@ module Reply = struct
 
   let result_of_command_outcome { success; error } =
     if success then Result.Ok () else Result.Error (error |? "")
+
+  type binding_modes = string list
+    [@@deriving of_yojson, show]
+
+  type config = {
+    config : string
+  } [@@deriving of_yojson, show]
+
+  let pp_config fmt config =
+    Format.pp_print_string fmt config
```

Comment:
I think we should be consistent: if we expose `type config = { config : string }` in the interface, then all functions should operate on the `config` type, without accessing the field themselves. It is a bit heavier for the user, but more future proof: we won't need to break the api if a field gets added.